app/routes/quotes/components/QuotePage.tsx

```diff
@@ -1,96 +1,110 @@
-  deleteQuote: (id: ID) => Promise<void>;
-  fetchAll: (args: {
-    query: {
-      approved: string;
-      ordering: string;
-    };
-    next?: boolean;
-  }) => Promise<void>;
-  showFetchMore: boolean;
-  currentUser: CurrentUser;
-  loggedIn: boolean;
-  addReaction: (args: {
-    emoji: string;
-    contentTarget: ContentTarget;
-  }) => Promise<void>;
-  deleteReaction: (args: {
-    reactionId: ID;
-    contentTarget: ContentTarget;
-  }) => Promise<void>;
-  fetching: boolean;
-  fetchEmojis: () => Promise<void>;
-  fetchingEmojis: boolean;
-  emojis: Emoji[];
-};
+const qsParamsParser = (search) => ({
+  approved:
+    qs.parse(search, {
+      ignoreQueryPrefix: true,
+    }).approved || 'true',
+  ordering: qs.parse(search, {
+    ignoreQueryPrefix: true,
+  }).ordering,
+});
+
 type Option = {
   label: string;
   value: string;
 };
+
 const filterRegDateOptions: Array<Option> = [
   {
     label: 'nyeste',
     value: '',
   },
   {
-    label: 'flest emojis',
+    label: 'flest reaksjoner',
     value: '?ordering=-reaction_count',
   },
 ];
-export default function QuotePage({
-  query,
-  quotes,
-  approve,
-  unapprove,
-  actionGrant,
-  deleteQuote,
-  fetchAll,
-  showFetchMore,
-  currentUser,
-  loggedIn,
-  addReaction,
-  deleteReaction,
-  emojis,
-  fetching,
-  fetchEmojis,
-  fetchingEmojis,
-}: Props) {
-  let errorMessage = undefined;
 
+type Props = {
+  loggedIn: boolean;
+};
+
+const QuotePage = ({ loggedIn }: Props) => {
+  const dispatch = useAppDispatch();
+
+  const { quoteId } = useParams();
+
+  const location = useLocation();
+
+  const { pagination } = useAppSelector((state) =>
+    selectPaginationNext({
+      endpoint: `/quotes/`,
+      query: qsParamsParser(location.search),
+      entity: 'quotes',
+    })(state)
+  );
+  const showFetchMore = pagination.hasMore;
+
+  const quotes = useAppSelector((state) => {
+    if (quoteId) {
+      return [selectQuoteById(state, quoteId)];
+    }
+
+    return selectQuotes(state, { pagination });
+  });
+
+  const fetching = useAppSelector((state) => state.quotes.fetching);
+
+  const query = qs.parse(location.search, {
+    ignoreQueryPrefix: true,
+  });
+
+  let errorMessage = undefined;
   if (quotes.length === 0 && !fetching) {
     errorMessage =
       query.approved === 'false'
         ? 'Ingen sitater venter på godkjenning.'
         : 'Fant ingen sitater. Hvis du har sendt inn et sitat venter det trolig på godkjenning.';
   }
 
+  const actionGrant = useAppSelector((state) => state.quotes.actionGrant);
+  const emojis = useAppSelector((state) => selectEmojis(state));
+  const fetchingEmojis = useAppSelector((state) => state.emojis.fetching);
+
+  useEffect(() => {
+    if (quoteId) {
+      dispatch(fetchQuote(quoteId));
+    } else {
+      dispatch(
+        fetchAll({
+          query: qsParamsParser(location.search),
+        })
+      );
+    }
+    dispatch(fetchEmojis());
+  }, [quoteId, location.search, dispatch]);
```

Comment:
```suggestion
  usePreparedEffect(
    '',
    () => {
      if (quoteId) {
        dispatch(fetchQuote(quoteId));
      } else {
        dispatch(
          fetchAll({
            query,
          })
        );
      }
      dispatch(fetchEmojis());
    },
    [quoteId, location.search, dispatch]
  );
  ```
 Use `usePreparedEffect` to eventually support SSR:)

Reply:
Yes, but I'll fix this in the fifth batch because I don't want to rebase and fix 4 different branches. There's some other ssr related stuff that needs to be delt with